### PR TITLE
Add json schema gen

### PR DESCRIPTION
**Pull request by TimTheBig**
This patch adds a json schema that gets generated on debug runs.
It also further documents tower.rs.

### CODE REVIEW

File: gladius_shared/Cargo.toml
Comment (mrhatman) [inline gladius_shared/Cargo.toml:1]: You should be able to only include this for debug builds

File: src/plotter/polygon_operations.rs
Comment (mrhatman) [inline src/plotter/polygon_operations.rs:1]: Lets not remove this. I understand its currently not used so feel free to mark as unused but I want it available incase it's needed in the future.

**Review by mrhatman — COMMENTED** [review]
Mostly looks good.

File: gladius_shared/Cargo.toml
Comment (TimTheBig) [reply gladius_shared/Cargo.toml:1]: I don't think this is possible without a feature flag, if that is ok I will add it soon.

File: src/plotter/polygon_operations.rs
Comment (TimTheBig) [reply src/plotter/polygon_operations.rs:1]: I have reverted this change, it is now resolved.

File: gladius_shared/Cargo.toml
Comment (TimTheBig) [reply gladius_shared/Cargo.toml:1]: This has been implemented.

**Pull request merged**